libraries/AP_Scripting/examples/rev_thr_on_land.lua

```diff
@@ -0,0 +1,181 @@
+      and (rc:get_pwm(RC_ARM_CHANNEL_:get()) > ARM_CHANNEL_HI_PWM_:get())
+      and (THROT_CHANNEL_:norm_input_dz() == 0)) then
+    if (RFND_READING() < THRESH_LAND_HEIGHT_:get()) then
+      if (COUNT_IN_RANGE_ > TIM_AFTER_TOUCHDN_) then
+        TIME_SEC_ = millis():tofloat()
+        return ENGAGE_REV_THRUST()
+      end
+      COUNT_IN_RANGE_ = COUNT_IN_RANGE_ + 1
+    else
+      COUNT_IN_RANGE_ = 0
+    end
+  end
+  return UPDATE, 100 -- check again in 10Hz
+end
+
+function RFND_READING()
+  if rangefinder:has_data_orient(RFND_ROT_DN_) then
+    return rangefinder:distance_cm_orient(RFND_ROT_DN_) * 0.01
+  end
+  return MAX_DIST_
+end
+
+function DISENGAGE_REV_THRUST()
+  THROT_CHANNEL_:set_override(REV_THRUST_IDLE_PWM_:get())
+  REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_LOW_PWM_)
+  gcs:send_text(4, "Reverse Thrust Disengaged")
+  IS_IN_REV_THRUST_ = false
+end
+
+function ENGAGE_REV_THRUST()
+  local groundspeed = ahrs:groundspeed_vector():length()
+
+  if (IS_THR_SUP_ and (vehicle:get_mode() == MODE_FBWA)) then
+    if (millis():tofloat() - TIME_SEC_ < TIME_THROT_WAIT_MS_:get()) then
+      THROT_CHANNEL_:set_override(REV_THRUST_IDLE_PWM_:get()) -- set idle throttle and prepare for reverse, letting the motors stop
+    else
+      REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_HIGH_PWM_)
+      if (REV_THRUST_AUX_SW_:get_aux_switch_pos() == 2) then
+        IS_THR_SUP_ = false
+        REV_THRUST_SW_ = true --set flag for reverse thrust active
+      end
+    end
+  end
+
+  if ((REV_THRUST_AUX_SW_:get_aux_switch_pos() == 2)
+    and (IS_THR_SUP_ == false)
+    and (REV_THRUST_SW_ == true)) then
+    TIME_SEC_ = millis():tofloat()
+    REV_THRUST_SW_ = false
+    IS_IN_REV_THRUST_ = true
+    REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_HIGH_PWM_)
+    gcs:send_text(4, "Reverse Thrust Engaged")
+  end
+
+  if (IS_IN_REV_THRUST_) then
+    if (groundspeed > MIN_GROUND_SPEED_:get()) then
+      if (REV_THRUST_AUX_SW_:get_aux_switch_pos() == 2) then
+        REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_HIGH_PWM_)
+        THROT_CHANNEL_:set_override(math.min(REV_THRUST_CUR_PWM_, REV_THRUST_HIGH_PWM_:get())) -- set rev thrust
+        REV_THRUST_CUR_PWM_ = REV_THRUST_CUR_PWM_ + math.floor(REV_THRUST_CHG_PWM_PS_:get() * CALLBACK_TIME_ / 1000)
+        if (millis():tofloat() - TIME_SEC_ > MAXTIME_REV_THRUST_APPLIED_MS_:get()) then
+          DISENGAGE_REV_THRUST()
+          return --stop the script
+        end
+      else
+        DISENGAGE_REV_THRUST()
+        return --stop the script
+      end
+    else
+      DISENGAGE_REV_THRUST()
+      return --stop the script
+    end
+  end
+  return ENGAGE_REV_THRUST, CALLBACK_TIME_
+end
+
+-- checking Vehicle Type and FW Version
+if (not(FWVersion:type() == 3)) then
+  gcs:send_text(4, "Rev_Throttle Script: must be running Plane FW")
+  return
+end
+
+if (AP_VERSION_NUM_ < 41) then
+  gcs:send_text(4, "Rev_Throttle Script: must be running Plane4.1 or higher")
```

Comment:
you could assert this it will cause a script error which will be harder to miss.